From c1bf266e1eba07136afce0ac7595dbfe74f395d1 Mon Sep 17 00:00:00 2001
From: Mat Kowalski <mko@redhat.com>
Date: Wed, 8 Jul 2026 20:09:49 +0200
Subject: [PATCH 22/26] MCO: point kube-vip at the node kubeconfig

kube-vip defaults to in-cluster config which static pods do not have;
k8s_config_file directs it to the mounted node kubeconfig.

Found during the first end-to-end BGP VIP install.
---
 manifests/on-prem/0010-kube-vip-api.yaml                        | 2 ++
 templates/master/00-master/on-prem/files/0010-kube-vip-api.yaml | 2 ++
 .../master/00-master/on-prem/files/0020-kube-vip-ingress.yaml   | 2 ++
 3 files changed, 6 insertions(+)

diff --git a/manifests/on-prem/0010-kube-vip-api.yaml b/manifests/on-prem/0010-kube-vip-api.yaml
index b48417c22..bc7510008 100644
--- a/manifests/on-prem/0010-kube-vip-api.yaml
+++ b/manifests/on-prem/0010-kube-vip-api.yaml
@@ -20,6 +20,8 @@ spec:
     env:
     - name: address
       value: "{{ onPremPlatformAPIServerInternalIP .ControllerConfig }}"
+    - name: k8s_config_file
+      value: "/etc/kubernetes/kubeconfig"
     - name: port
       value: "6443"
     - name: cp_enable
diff --git a/templates/master/00-master/on-prem/files/0010-kube-vip-api.yaml b/templates/master/00-master/on-prem/files/0010-kube-vip-api.yaml
index fecd40ca6..5ec9bcb07 100644
--- a/templates/master/00-master/on-prem/files/0010-kube-vip-api.yaml
+++ b/templates/master/00-master/on-prem/files/0010-kube-vip-api.yaml
@@ -24,6 +24,8 @@ contents:
         env:
         - name: address
           value: "{{ onPremPlatformAPIServerInternalIP . }}"
+        - name: k8s_config_file
+          value: "/etc/kubernetes/kubeconfig"
         - name: port
           value: "6443"
         - name: cp_enable
diff --git a/templates/master/00-master/on-prem/files/0020-kube-vip-ingress.yaml b/templates/master/00-master/on-prem/files/0020-kube-vip-ingress.yaml
index 57f116a52..dec6ff5c5 100644
--- a/templates/master/00-master/on-prem/files/0020-kube-vip-ingress.yaml
+++ b/templates/master/00-master/on-prem/files/0020-kube-vip-ingress.yaml
@@ -24,6 +24,8 @@ contents:
         env:
         - name: address
           value: "{{ onPremPlatformIngressIP . }}"
+        - name: k8s_config_file
+          value: "/etc/kubernetes/kubeconfig"
         - name: port
           value: "443"
         - name: cp_enable
